test/e2e_test.go

```diff
@@ -0,0 +1,166 @@
+// +build e2e
+
+package test
+
+import (
+	"context"
+	"github.com/Shopify/sarama"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+const (
+	httpUrlPrefix   = "http://localhost:8080"
+	metricsEndpoint = "/metrics"
+)
+
+/* test checks for following:
+*  the presence of canary topic,
+*  liveliness of topic (messages being produced),
+*/
+func TestCanaryTopicLiveliness(t *testing.T) {
+	log.Println("TestCanaryTopic test starts")
+
+	// setting up timeout and handler for consumer group, there is no need to wait much longer then is the retention time.
+	timeout := time.After(time.Second * 10)
+	handler := exampleConsumerGroupHandler{}
+	handler.mutexWritePartitionPresence = &sync.Mutex{}
+	handler.consumingDone = make(chan bool)
```

Comment:
cannot do this initialization inside the above declaration ?

Reply:
I have created constructor for handler, and moved timeout to selector, but besides that timeout must stay within method.